### ISSUE TITLE
pkg.go.dev: lower score of mismatched module path (non-canonical paths or forks)

### ISSUE THREAD

**Issue by hhrutter** [golang/go#37187]
Hi!

Thank you for pkg.go.dev.
Great and very useful site!

I am the maintainer of the repository previously known as:
https://github.com/hhrutter/pdfcpu

We rolled pdfcpu into an organization some time ago and it is now located at:
https://github.com/pdfcpu/pdfcpu

When I search for `pdfcpu` the first couple of hits I get are for the old location of the repo and this is no good.

I am hoping there is a way to prioritize packages for search terms where both an organization and the repository name match over others.

Simply put if I search for `pdfcpu` I expect the first hits to point to:
https://github.com/pdfcpu/pdfcpu

**Comment by bcmills** [comment]
> I am hoping there is a way to prioritize packages for search terms where both an organization and the repository name match over others.

For this particular problem, I would prefer a different (and more general) approach: we should prioritize modules whose declared module paths match the path at which they were fetched over modules whose paths do not match.

A module with a mismatched path can only be used in a `replace` directive replacing the module with the declared path, so mismatched paths are always either non-canonical paths or forks — either way, they should be ranked lower than the corresponding canonical path.

(CC @julieqiu)

```
example.com$ go get -d github.com/hhrutter/pdfcpu@latest
go: downloading github.com/hhrutter/pdfcpu v0.3.2
go: github.com/hhrutter/pdfcpu latest => v0.3.2
go get: github.com/hhrutter/pdfcpu@v0.3.2: parsing go.mod:
        module declares its path as: github.com/pdfcpu/pdfcpu
                but was required as: github.com/hhrutter/pdfcpu
```

**Comment by hhrutter** [comment]
There is still a forward going on from the old to the new repo url.
I need to check if I can remove that.
Other than that 👍

**Comment by julieqiu** [comment]
We worked on a fix for handling mismatched paths, and it seems to have address the search for `pdfcpu`.

![image](https://user-images.githubusercontent.com/5943521/75591970-1c2f4e80-5a4f-11ea-9271-86640478a617.png)

Only the latest version of the canonical path will now show up in search results.

The logic is similar to what @bcmills described in https://github.com/golang/go/issues/37187#issuecomment-585262500.

Closing this issue.
